### DISCUSSION TITLE
Support for 915mhz LoRaWAN modules ?

### DISCUSSION THREAD

**Discussion by fiveangle** [PlummersSoftwareLLC/NightDriverStrip#295] [Q&A]
I'm considering using NightDriver as the brains for my Burning Man bicycle project where we have up to 6 bikes ridden together and synchronized to each other via a master node (either running on the ESP32 itself or a separate MCU/device if necessary). My understanding is that NightDriver supports synchronization via TCP/IP over a separate Wi-Fi AP (ad-hoc zeroconf not supported).

Does anyone know of a project that has supplanted the Wi-Fi client functionality of NightDriver with a LoRaWAN stack ?

I've not tested directly but word on the street is that the 2.4Ghz Wi-Fi channels are so congested at Burning Man that IoT devices sometimes cant go more than a few meters before getting buried in noise and losing sync to the AP. I'm researching possible mitigations like finding appropriate ESP32 module with high-gain antenna option, optimizing Wi-Fi network stack for high noise env (long preamble, very high beacon rate, minimum TCP packet size, etc) but if the graybeards of BM lighting projects say "none of that works once you get out there) I am considering a 900Mhz LoRaWAN solution to ensure functionality in all conditions and at least several hundred feet LoS. 

Anyone tried implementing NightDriver over 900Mhz LoRaWAN with any of the LoRaWAN ESP32 modules ? (Like https://a.co/d/2FlOSQX )

Thanks !

-=dave

**Comment by davepl** [comment]
No, I’ve never bought a LoRa chip!  I know next to nothing about it, but if the connection mechanism is at all orthogonal with WiFi, you could likely adapt the code pretty quickly.

I’d start with the basic LEDSTRIP project config and see if you can get that working!  It might “just work” with a bit of conversion to LoRa.

- Dave

**Comment by zephyrr** [comment]
There's a huge difference in bandwidth between LoRa and WiFi.  I'm not sure how that would affect synchronization.

You may not need LoRaWan tho, LoRa may be enough, unless you really need a gateway.

It won't avoid the congestion, but for that range you might also look into using ESP NOW protocol on the ESP's for synch.
